### DISCUSSION TITLE
How to ignore dependency's Python version constrain

### DISCUSSION THREAD

**Discussion by PaleNeutron** [python-poetry/poetry#9554] [Q&A]
I'm the maintainer of `dataframe_image` and I'm migrating my build system to poetry.

But a dependency  `html2image`'s required python version is ">=3.6,<4.0", I think the `<4.0` is not necessary. And if I add it to my dependency, all package depends on `dataframe_image` will have to do the same thing. Just like a virus.

Is there any way to ignore the  `<4.0` dependency?

**Comment by radoering** [comment]
I do not think that is possible for now but it will be possible with #9135 where you can distinguish between the Python constraint for your metadata and the Python constraint for locking dependencies.

**Comment by dimbleby** [comment]
You can make a merge request at htmlimage to relax the python requirement

**Reply by PaleNeutron** [reply]
But `html2image` add `<4.0` for the same reason. I can not push every package on the dependency chain to fix that.

And I found more and more packages affected by this settings. There should be a way to stop that at my package not just fix the source.

**Reply by dimbleby** [reply]
> But html2image add <4.0 for the same reason

nonsense.  `python = ">=3.6"` will work just fine in their project definition.
